### ISSUE TITLE
Better error message for related-field bugs

### ISSUE THREAD

**Issue by ienissei** [plk/biblatex#395]
For legal citation, I use many entries in `related` fields, including entires that relate to each other. It works while you have two entires pointing back to one another – it breaks when you point to a third entry.

For instance:

```
@book{ Book1,
       title = {Book 1},
       related = {Article1} }
@article{ Article1,
       title = {Article 1},
       related = {Book1} }% Switching to {Book2} fails at biber level
```

The limitation is not annoying in the least (related fields are such a blessing already), but I was wondering whether it would be possible to make the error message less cryptic? E.g. say it comes from a related-field or even state which one it chokes on?

**Comment by plk** [comment]
Can you give a MWE for this? I can't reproduce it at the moment. This should work I think.

**Comment by ienissei** [comment]
Sorry, hadn't tested it properly on the fly. It took several tries to reproduce it, but this MWE should fail properly. If you remove the extra cite-key, it should work again (trash all files before trying to crash it again).

```
\documentclass{article}

\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@article{ Cass:2013.10.17,
	Author = {Author},
	Related = {Comm:2013.10.17} }
@article{ Comm:2013.10.17,
	Author = {Author},
	Related = {Cass:2013.10.17, Cass:2009.01.29} }
\end{filecontents}

\usepackage[style=authortitle,backend=biber]{biblatex}
\addbibresource{\jobname.bib}

\begin{document}

\cite{Cass:2013.10.17}

\end{document}
```

Here is the error message: `Can't call method "get_field" on an undefined value at /var/folders/lp/h263wr2148151k634_jzld6c0000gn/T/par-4570697374696e65/cache-d000f62e9d0699f74e0990114d67a800379ec7e7/inc/lib/Biber.pm line 3263.`

**Comment by plk** [comment]
A quick question - does `Cass:2009.01.29` exist in the real example?

**Comment by ienissei** [comment]
The first time I had that issue, the entry existed but the configuration was more complex (and I wouldn't be able to reproduce it, I bypassed it). Then I mostly had the issue with typos in citekeys – which is why I was wondering whether it would be possible / simple to just make the error message easier to decode. In a large bibliography this is a killer.

**Comment by plk** [comment]
I'm glad you reported this - it was a long-standing bug in the routines to detect and deal with missing entries in nested entry relationships. It is now fixed in biber 2.5 DEV version on Sourceforge and your test case no longer crashes but emits a warning that `Cass:2009.01.29` doesn't exist. If you want to try it, you'll also need to be using the biblatex 3.4 developement version, also on Sourceforge.

**Comment by ienissei** [comment]
Thank you very much for fixing it so quickly.
